File: bindings/python/example_basic.py
#!/usr/bin/env python3
"""
Basic example of using matgto-serge from Python

This example shows how to:
1. Create a proxy in record mode
2. Start recording HTTP/WebSocket interactions
3. Stop recording and save to cassette
"""

import sys
import requests
from matgto_serge import create_proxy, ProxyMode

def main():
    print("🎬 matgto-serge Python Example - Basic Recording")
    print("=" * 50)

    # Create proxy instance
    print("\n1️⃣ Creating proxy...")
    proxy = create_proxy("./cassettes")
    proxy = proxy.with_port(8888)
    proxy = proxy.with_mode(ProxyMode.RECORD)
    print(f"   ✅ Proxy created on port {proxy.port()}")

    # Start recording
    print("\n2️⃣ Starting recording...")
    cassette_name = "python-example"
    proxy.start_recording(cassette_name)
    print(f"   ✅ Recording to cassette: {cassette_name}")

    # Make HTTP requests through proxy
    print("\n3️⃣ Making HTTP requests through proxy...")
    print("   (In a real scenario, configure your HTTP client to use proxy)")
    print("   Example: requests.get('https://httpbin.org/get', proxies={'http': 'http://localhost:8888'})")

    # Stop recording
    print("\n4️⃣ Stopping recording...")
    proxy.stop_recording()
    print(f"   ✅ Cassette saved: ./cassettes/{cassette_name}.json")

    # Shutdown proxy
    print("\n5️⃣ Shutting down proxy...")
    proxy.shutdown()
    print("   ✅ Proxy stopped")

    print("\n" + "=" * 50)
    print("✅ Example complete!")

if __name__ == "__main__":
    try:
        main()
    except Exception as e:
        print(f"\n❌ Error: {e}", file=sys.stderr)
        sys.exit(1)

File: bindings/php/example_basic.php
<?php
/**
 * Basic example of using matgto-serge from PHP
 *
 * This example shows how to:
 * 1. Create a proxy in record mode
 * 2. Start recording HTTP/WebSocket interactions
 * 3. Stop recording and save to cassette
 */

require_once __DIR__ . '/MagnetoProxy.php';

use MatgtoSerge\MagnetoProxy;
use MatgtoSerge\ProxyMode;

echo "🎬 matgto-serge PHP Example - Basic Recording\n";
echo str_repeat("=", 50) . "\n\n";

try {
    // Create proxy instance
    echo "1️⃣ Creating proxy...\n";
    $proxy = new MagnetoProxy("./cassettes");
    $proxy->withPort(8888)
          ->withMode(ProxyMode::Record);
    echo "   ✅ Proxy created on port " . $proxy->getPort() . "\n";

    // Start recording
    echo "\n2️⃣ Starting recording...\n";
    $cassetteName = "php-example";
    $proxy->startRecording($cassetteName);
    echo "   ✅ Recording to cassette: $cassetteName\n";

    // Make HTTP requests through proxy
    echo "\n3️⃣ Making HTTP requests through proxy...\n";
    echo "   (Configure your HTTP client to use proxy)\n";
    echo "   Example with Guzzle:\n";
    echo "   \$client = new \\GuzzleHttp\\Client(['proxy' => 'http://localhost:8888']);\n";
    echo "   \$response = \$client->get('https://httpbin.org/get');\n";

    // Simulate some work
    sleep(1);

    // Stop recording
    echo "\n4️⃣ Stopping recording...\n";
    $proxy->stopRecording();
    echo "   ✅ Cassette saved: ./cassettes/{$cassetteName}.json\n";

    // Shutdown proxy
    echo "\n5️⃣ Shutting down proxy...\n";
    $proxy->shutdown();
    echo "   ✅ Proxy stopped\n";

    echo "\n" . str_repeat("=", 50) . "\n";
    echo "✅ Example complete!\n";

} catch (Exception $e) {
    echo "\n❌ Error: " . $e->getMessage() . "\n";
    exit(1);
}
